pgex/examples/platformer_collisions/main.py

```diff
@@ -0,0 +1,125 @@
+"""
+This file is a part of the 'Pygame Examples (pgex)' source code.
+The source code is distributed under the MIT license.
+"""
+
+import asyncio
+
+import pygame
+
+from . import common
+
+
+def get_colliding_tiles(tiles: list, player_rect: pygame.Rect) -> list:
+    """
+    Returns a list of tiles the player is currently colliding with
+    """
+    return_tiles = []
+    for tile in tiles:
+        if player_rect.colliderect(tile):
+            return_tiles.append(tile)
+
+    return tiles
+
+
+def calculate_rect(
+    movement: list, player_rect: pygame.Rect, map_tiles: list
+) -> pygame.Rect:
+    """
+    Calculates the Rect of the player based on their movement and the surrounding tiles
+    """
+    player_rect.x += movement[0]
+    tiles = get_colliding_tiles(map_tiles, player_rect)
+    for tile in tiles:
+        if player_rect.colliderect(tile):
+            if movement[0] > 0:
+                player_rect.right = pygame.Rect(tile[0], tile[1], tile[2], tile[3]).left
+            if movement[0] < 0:
+                player_rect.left = pygame.Rect(tile[0], tile[1], tile[2], tile[3]).right
+
+    common.is_on_ground = False
+    player_rect.y += movement[1]
+    tiles = get_colliding_tiles(map_tiles, player_rect)
+    for tile in tiles:
+        if player_rect.colliderect(tile):
+            if movement[1] > 0:
+                player_rect.bottom = pygame.Rect(tile[0], tile[1], tile[2], tile[3]).top
+                common.is_on_ground = True
+                common.y_velocity = 3
+            if movement[1] < 0:
+                player_rect.top = pygame.Rect(tile[0], tile[1], tile[2], tile[3]).bottom
+
+    return player_rect
+
+
+def player_movement(key_presses: dict, player_rect: pygame.Rect) -> pygame.Rect:
+    """
+    Handles all code realting to the movement of the player
+    """
+
+    player_movement = [0, common.y_velocity]
+    if key_presses["a"]:
+        player_movement[0] -= 4
+    if key_presses["d"]:
+        player_movement[0] += 4
+
+    if common.y_velocity < 10:
+        common.y_velocity += 0.5
```

Comment:
Same, consider extracting these raw values inside of `common` module so the readers will directly know by seeing the name of the constant what is the meaning behind the values and it would be really easy to tweak the physic of the game (all the values would be defined at the same place).